csatar-plugins/csatar/components/CreateFrontendAccounts.php

```diff
@@ -0,0 +1,300 @@
+                throw new ValidationException(['ecset_code' => Lang::get('csatar.csatar::lang.plugin.component.createFrontendAccounts.validationExceptions.emailEcsetCodeMissMatch')]);
+            }
+
+            if (!$this->canRegister()) {
+                throw new ApplicationException(Lang::get(/*Registrations are currently disabled.*/'rainlab.user::lang.account.registration_disabled'));
+            }
+
+            if ($this->isRegisterThrottled()) {
+                throw new ApplicationException(Lang::get(/*Registration is throttled. Please try again later.*/'rainlab.user::lang.account.registration_throttled'));
+            }
+
+            if ($this->loginAttribute() !== UserSettings::LOGIN_USERNAME) {
+                unset($rules['username']);
+            }
+
+            if ($ipAddress = Request::ip()) {
+                $data['created_ip_address'] = $data['last_ip_address'] = $ipAddress;
+            }
+
+            Event::fire('rainlab.user.beforeRegister', [&$data]);
+
+            $automaticActivation = UserSettings::get('activate_mode') == UserSettings::ACTIVATE_AUTO;
+            $user = Auth::register($data, $automaticActivation);
+
+            Event::fire('rainlab.user.register', [$user, $data]);
+
+            $scout->user_id = $user->id;
+            $scout->save();
+
+            if ($automaticActivation) {
+                Auth::login($user, $this->useRememberLogin());
+            }
+
+            return Redirect::to('/tag/' . $ecsetCode);
+        }
+        catch (Exception $ex) {
+            if (Request::ajax()) throw $ex;
+            else Flash::error($ex->getMessage());
+        }
+    }
+
+    public function onCreateFrontendAccounts(){
+
+        $data = Input::get('scouts');
+
+        $scouts = Scout::whereIn('id', $data)->get();
+
+        foreach ($scouts as $scout) {
+
+            if(!empty($scout->user_id)){
+                $this->messages['errors'][$scout->id] =
+                    Lang::get('csatar.csatar::lang.plugin.component.createFrontendAccounts.messages.scoutAlreadyHasUserAccount',
+                        ['name' => $scout->getFullName() ], 'hu');
+            }
+
+            if(!empty($scout->email) && empty($scout->user_id)){
+                $this->register($scout);
+            }
+
+            if(empty($scout->email)){
+                $this->messages['errors'][$scout->id] =
+                    Lang::get('csatar.csatar::lang.plugin.component.createFrontendAccounts.messages.scoutHasNoEmail',
+                        ['name' => $scout->getFullName() ]);
+            }
+        }
+
+        return [
+            '#messageList' => $this->renderPartial('@messages',)
+        ];
+    }
+
+    /**
+     * Register the user
+     */
+    public function register($scout)
+    {
+        try {
+            if (!$this->canRegister()) {
+                throw new ApplicationException(Lang::get(/*Registrations are currently disabled.*/'rainlab.user::lang.account.registration_disabled'));
```

Comment:
Avoid comment blocks (more instances in this file)

Reply:
Removed.